src/spec-node/featuresCLI/testCommandImpl.ts

```diff
@@ -103,9 +105,105 @@ async function runGlobalFeatureTests(args: FeaturesTestCommandInput, testResults
 	return testResults;
 }
 
+// Executes the same Feature twice with randomized options to ensure Feature can be installed >1.
+async function runDuplicateTest(args: FeaturesTestCommandInput, feature: string, testResults: TestResult[] = []): Promise<TestResult[]> {
+	const { collectionFolder, cliHost } = args;
+	const scenarioName = `${feature} executed twice with randomized options`;
+
+	const featureTestFolder = path.join(collectionFolder, 'test', feature);
+	const testFileName = 'duplicate.sh';
+	const testFilePath = path.join(featureTestFolder, testFileName);
+	if (!(await cliHost.isFile(testFilePath))) {
+		log(`Skipping duplicate test for ${feature} because '${testFilePath}' does not exist.`, { prefix: '⚠️', });
+		return testResults;
+	}
+
+	//Read Feature's metadata
+	const featureMetadata = await readFeatureMetadata(args, feature);
+	const options = featureMetadata.options || {};
+
+	// For each possible option, generate a random value for each Feature
+	const randomizedOptions: { [key: string]: string | boolean } = {};
+	Object.entries(options).forEach(([key, value]) => {
+		if (value.type === 'boolean') {
+			randomizedOptions[key] = !value.default;
+		}
+		if (value.type === 'string' && 'proposals' in value && value?.proposals?.length) {
+			const randomIndex = Math.floor(Math.random() * value.proposals.length);
```

Comment:
Using randomized options might result in unexpected failures. Not sure if these will been seen as helpful. Maybe add command line option to enable/disable randomization and consider having it disabled by default to avoid unhelpful failures.

Reply:
Yea I agree. My goal is making it easy to test installing two different "versions" of a given Feature.  The alternatives are:

- Two of the exact same Feature with same options (less interesting)
- One Feature with default value and one with a non-default but static set of options.

Reply:
I've added a flag that can be passed to add in randomness, otherwise it will be non-default, but predictable values (unless the devcontainer-feature.json is updated)